File: back/kotlin/app/connections/config/Environment.kt
import java.io.File
import java.util.*

object EnvironmentLoader {
    private val env = loadEnv()

    fun getSessionSecret(): String = env["SESSION_SECRET"] ?: ""
    fun getRedisHost(): String = env["REDIS_HOST"] ?: ""

    private fun loadEnv(): Map<String, String> {
        val env = mutableMapOf<String, String>()
        val currentDir = File(System.getProperty("user.dir"))
        val envFile = File(currentDir.parentFile?.parentFile, ".env")

        if (envFile.exists()) {
            envFile.readLines().forEach { line ->
                val (key, value) = line.split("=", limit = 2)
                env[key] = value
            }
        }

        return env
    }
}

File: back/node/app/connections/config/environment.ts
import dotenv, {DotenvConfigOutput} from 'dotenv';

const result: DotenvConfigOutput = dotenv.config();
if (result.error) {
    console.log(`Dotenv Config Error ${result.error}`);
}

const SESSION_SECRET = process.env.SESSION_SECRET;
const REDIS_HOST = process.env.REDIS_HOST;

export {SESSION_SECRET, REDIS_HOST};